pygmt/helpers/utils.py

```diff
@@ -66,7 +71,7 @@ def data_kind(data, x=None, y=None, z=None):
     if data is None and (x is None or y is None):
         raise GMTInvalidInput("Must provided both x and y.")
 
-    if isinstance(data, str):
+    if isinstance(data, (str, pathlib.Path)):
```

Comment:
What will happen if people use `pathlib.PurePath` or `path.PurePosixPath`?

Reply:
The answer at https://stackoverflow.com/questions/58647584/how-to-test-if-object-is-a-pathlib-path/58966089#58966089 suggests using `isinstance(pathlib.PurPath)`. @aitorres, could you perhaps see if this works?

```suggestion
    if isinstance(data, (str, pathlib.PurePath)):
```

Also, it might be a good idea to test that `PureWindowsPath` and `PurePosixPath` objects work on Windows and macOS/Linux respectively. Let us know if you're able to update the unit test to do that. If not, we can also just ignore it.

Reply:
Gentle ping @aitorres to see if you're still available to finish up this PR! If not, we can get one of the maintainers to finish this up in the next week or so.

Reply:
Really sorry for the delay! I've just added and pushed the changes to use `PurePath`, and added two tests that specifically use `PureWindowsPath` and `PurePosixPath`.